src/config.py

```diff
@@ -39,19 +38,27 @@ def _ingress_proxy_conf(self) -> dict[str, str]:
 
     @property
     def _s3_conf(self) -> dict[str, str]:
-        return (
-            {
+        conf = {}
+        if self.s3_connection_info:
+            conf = {
                 "spark.hadoop.fs.s3a.endpoint": self.s3_connection_info.endpoint
                 or "https://s3.amazonaws.com",
                 "spark.hadoop.fs.s3a.access.key": self.s3_connection_info.access_key,
                 "spark.hadoop.fs.s3a.secret.key": self.s3_connection_info.secret_key,
                 "spark.eventLog.dir": self.s3_connection_info.log_dir,
                 "spark.history.fs.logDirectory": self.s3_connection_info.log_dir,
                 "spark.hadoop.fs.s3a.aws.credentials.provider": "org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider",
+                "spark.hadoop.fs.s3a.connection.ssl.enabled": "false",
             }
-            if self.s3_connection_info
-            else {}
-        )
+            if self.s3_connection_info.tls_ca_chain:
+                conf.update(
+                    {
+                        "spark.executor.extraJavaOptions": "-Djavax.net.ssl.trustStore=/opt/spark/conf/truststore.jks -Djavax.net.ssl.trustStorePassword=changeit",
+                        "spark.driver.extraJavaOptions": "-Djavax.net.ssl.trustStore=/opt/spark/conf/truststore.jks -Djavax.net.ssl.trustStorePassword=changeit",
+                        "spark.hadoop.fs.s3a.connection.ssl.enabled": "true",
```

Comment:
not sure about this. the AWS and Google root certs are already in the truststore. I don't think you can assume that ssl is only enabled when there's a ca-chain provided